Add support for latest Terraform version 0.10.7

mpmsimo opened the pull request:
- Added Terraform versions 0.10.7 which has been set to the default version

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/hootsuite/atlantis?pullRequest=159) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/hootsuite/atlantis?pullRequest=159) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/hootsuite/atlantis?pullRequest=159) it.</sub>

lkysow commented:
@mpmsimo good call on the terraform version upgrade, why did you close the PR?

mpmsimo commented:
@lkysow I was making some changes within a fork of in order to customize Atlantis and realized the original scope was too large for this pull request, as I was introducing interactions with technology that might not be supported.

I can certainly update the PR in order to have the latest versions: 0.10.2 is currently what we are enforcing in our environments. Having the option to switch to the latest version 0.10.7 is also a nice to have.

mpmsimo commented:
The scope of this PR did expand a bit for our use case, so I've removed those additions to keep it an update to the terraform version in the Dockerfile itself.

Added 0.10.2 and 0.10.7 as versions in addition to the older options. Are there any versions that should be included as well? Are you guys using a version that's mentioned within the list?

lkysow commented on Dockerfile:
```diff
@@ -28,9 +28,9 @@ RUN apk add --no-cache ca-certificates gnupg curl git unzip bash openssh libcap
+ENV DEFAULT_TERRAFORM_VERSION=0.10.7
 
-RUN AVAILABLE_TERRAFORM_VERSIONS="0.8.8 0.9.11 0.10.0" && \
+RUN AVAILABLE_TERRAFORM_VERSIONS="0.8.8 0.9.11 0.10.0 0.10.2 0.10.7" && \
```
we would like to only have the latest versions of each point release, so this should be `0.8.8 0.9.11 0.10.7`

lkysow commented:
@mpmsimo we need you to sign the CLA as well to accept the pull request
